openconnect_sso/app.py

```diff
@@ -179,3 +181,10 @@ def run_openconnect(auth_info, host, proxy, args):
 
     logger.debug("Starting OpenConnect", command_line=command_line)
     return subprocess.run(command_line).returncode
+
+
+def handle_disconnect(command):
+    if command:
+        logger.info(f"Running {command!r} on shutdown...")
+        argv = [str(Path(command).expanduser())]
+        return subprocess.run(argv, timeout=5).returncode
```

Comment:
IMHO this command does not support arguments right now, is that right? If you set `shell=True` in `subprocess.run` then you could use a command line with arguments. (or you can split the command line with `shlex.quote` instead)

Reply:
Correct. Didn't need arguments in my specific use case, but it's certainly needed in a more general version. Since we're going via a string in the config file, I'll use `shell=True` here. Although I'm usually not a fan of `shell=True`, we should be fine as all the input ultimately comes from the (trusted) user. Alternatively, we could use a list of string args in `config.toml`, but that doesn't feel very natural.